src/Request.php

```diff
@@ -722,7 +722,10 @@ private function is_valid_http_content_type(): bool {
 			return false;
 		}
 
-		return 0 === stripos( $content_type, 'application/json' );
+		/** 
+  		 * Allow graphql to validate custom content types for HTTP POST requests
+     		 */
+		return apply_filters( 'graphql_is_valid_http_content_type', 0 === stripos( $content_type, 'application/json' ), $content_type, $_SERVER['REQUEST_METHOD'] );
```

Comment:
1. For maintainability, I'd recommend moving the `$content_type` check to an `$is_valid` var passed to the filter, instead of inlining and obscuring the default.
2. We can remove `$_SERVER` from the filter, it's available to any callable w.o a param.
3. We should backfill the doc-block params before release